pkg/reconciler/pipeline/controller.go

```diff
@@ -725,6 +726,20 @@ func (r *pipelineReconciler) updateDesiredState(ctx context.Context, pl *dfv1.Pi
 }
 
 func (r *pipelineReconciler) resumePipeline(ctx context.Context, pl *dfv1.Pipeline) (bool, error) {
+
+	// reset pause timestamp
+	if pl.GetAnnotations()[dfv1.KeyPauseTimestamp] != "" {
+		pl.GetAnnotations()[dfv1.KeyPauseTimestamp] = ""
```

Comment:
This means the annotation is not nil, why don't use `del()`?

Reply:
MergePatchType only merges the patch with the current object, it doesn't replace the `annotations` map. We would need to use `StrategicMergePatch` which isn't supported for this.

Reply:
Found that you can delete this annotation using this JSON patch: 
`[{"op": "remove", "path": "/metadata/annotations/numaflow.numaproj.io~1pause-timestamp"}]`

Implemented and added patch string to `const.go`

Reply:
This is nice!